### PR TITLE
feat: add seed cli command and add missing api handlers

### PR DESCRIPTION
**Pull request by Chief-Rishab**
- Seed CLI command bootstraps an entire org structure with project, resource, users and custom permission and roles in shield db to test the shield deployment. 
- This PR also adds missing handlers for Creating platform wide roles and deleting them, fixes minor bugs in user repo

### CODE REVIEW

File: internal/api/v1beta1/role.go [hunk 1]
```diff
@@ -81,6 +81,49 @@ func (h Handler) ListRoles(ctx context.Context, request *shieldv1beta1.ListRoles
 	return &shieldv1beta1.ListRolesResponse{Roles: roles}, nil
 }
 
+func (h Handler) CreateRole(ctx context.Context, request *shieldv1beta1.CreateRoleRequest) (*shieldv1beta1.CreateRoleResponse, error) {
```
Comment (kushsharma) [inline internal/api/v1beta1/role.go:84]: We need to protect this endpoint in `interceptor`

File: internal/api/v1beta1/role.go [hunk 2]
```diff
@@ -81,6 +81,49 @@ func (h Handler) ListRoles(ctx context.Context, request *shieldv1beta1.ListRoles
...
+	logger := grpczap.Extract(ctx)
+	metaDataMap, err := metadata.Build(request.GetBody().GetMetadata().AsMap())
```
Comment (kushsharma) [inline internal/api/v1beta1/role.go:86]: What happens when `request.GetBody().GetMetadata()` is nil? Let's add a test for this?

File: internal/api/v1beta1/role.go [hunk 3]
```diff
@@ -201,8 +244,31 @@ func (h Handler) UpdateOrganizationRole(ctx context.Context, request *shieldv1be
 	return &shieldv1beta1.UpdateOrganizationRoleResponse{Role: &rolePB}, nil
 }
 
+func (h Handler) DeleteRole(ctx context.Context, request *shieldv1beta1.DeleteRoleRequest) (*shieldv1beta1.DeleteRoleResponse, error) {
```
Comment (kushsharma) [inline internal/api/v1beta1/role.go:247]: protect this endpoint

File: cmd/seed.go [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+	}
+
+	bindFlagsFromClientConfig(cmd)
+
+	return cmd
+}
+
+// create compute/order namespace and custom roles and permissions
+func createCustomRolesAndPermissions(client shieldv1beta1.AdminServiceClient) error {
+	ctx := metadata.NewOutgoingContext(context.Background(), metadata.New(map[string]string{
```
Comment (kushsharma) [inline cmd/seed.go:80]: We can pass the context from command

File: cmd/seed.go [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+	mockHumanUser []byte
+	//go:embed seed/organization.json
+	mockOrganization []byte
+	//go:embed seed/project.json
+	mockProject []byte
+	//go:embed seed/resource.json
+	mockResource []byte
+
+	orgAdminEmail  = "sampleAdmin@raystack.org"
+	identityHeader = "X-Shield-Email"
```
Comment (kushsharma) [inline cmd/seed.go:30]: `identityHeader` can also come from config/flag as it can change based on deployments?

File: cmd/seed.go [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+	//go:embed seed/user.json
+	mockHumanUser []byte
+	//go:embed seed/organization.json
+	mockOrganization []byte
+	//go:embed seed/project.json
+	mockProject []byte
+	//go:embed seed/resource.json
+	mockResource []byte
+
+	orgAdminEmail  = "sampleAdmin@raystack.org"
```
Comment (kushsharma) [inline cmd/seed.go:29]: `orgAdminEmail` looks very general to shield, we should name it specific for seed command. For example: `sampleSeedEmail`? Also camel casing email doesn't look good, how about `sample.admin@raystack.org`

File: cmd/seed.go [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+// create compute/order namespace and custom roles and permissions
+func createCustomRolesAndPermissions(client shieldv1beta1.AdminServiceClient) error {
+	ctx := metadata.NewOutgoingContext(context.Background(), metadata.New(map[string]string{
+		identityHeader: orgAdminEmail,
+	}))
+
+	var permissionBodies []*shieldv1beta1.PermissionRequestBody
+	if err := json.Unmarshal(mockCustomPermissions, &permissionBodies); err != nil {
+		fmt.Println("Error unmarshaling JSON :", err)
+		return err
```
Comment (kushsharma) [inline cmd/seed.go:87]: this will print error twice right?

File: cmd/seed.go [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+	}
+
+	// create custom permission
+	if _, err := client.CreatePermission(ctx, &shieldv1beta1.CreatePermissionRequest{
+		Bodies: permissionBodies,
+	}); err != nil {
+		return fmt.Errorf("failed to create custom permission: %w", err)
+	}
+
+	fmt.Printf("created custom permissions for compute order service: %s, %s, %s, %s, %s\n", "create", "get", "update", "delete", "list")
```
Comment (kushsharma) [inline cmd/seed.go:97]: Permissions are read from json, hardcoding this log can go out of sync. Should we also print this based on json?

File: cmd/seed.go [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+
+	fmt.Printf("created custom roles %s and %s\n", "compute_order_manager", "compute_order_viewer")
+
+	return nil
+}
+
+func bootstrapData(client shieldv1beta1.ShieldServiceClient) error {
+	ctxOrgAdminAuth := metadata.NewOutgoingContext(context.Background(), metadata.New(map[string]string{
+		identityHeader: orgAdminEmail,
+	}))
```
Comment (kushsharma) [inline cmd/seed.go:120]: instead of doing this in every method, we can just do it in `RunE` func?

File: cmd/seed.go [hunk 10]
```diff
@@ -0,0 +1,196 @@
...
+
+	// create (human) user for org admin
+	var userBody *shieldv1beta1.UserRequestBody
+	if err := json.Unmarshal(mockHumanUser, &userBody); err != nil {
+		return fmt.Errorf("failed to unmarshal user body: %w", err)
+	}
+	userResp, err := client.CreateUser(ctxOrgAdminAuth, &shieldv1beta1.CreateUserRequest{
+		Body: userBody,
+	})
+
```
Comment (kushsharma) [inline cmd/seed.go:130]: nit: unnecessary new line

File: cmd/seed.go [hunk 11]
```diff
@@ -0,0 +1,196 @@
...
+		return fmt.Errorf("failed to create sample user: %w", err)
+	}
+
+	fmt.Printf("created user with email %s in shield\n", userResp.User.Email)
+
+	// create organization
+	var orgBody *shieldv1beta1.OrganizationRequestBody
+	if err := json.Unmarshal(mockOrganization, &orgBody); err != nil {
+		fmt.Println("Error unmarshaling JSON :", err)
+		return err
```
Comment (kushsharma) [inline cmd/seed.go:141]: merge these if needed

File: cmd/seed/project.json [hunk 12]
```diff
@@ -0,0 +1,8 @@
+{
+  "name": "compute-guard-project",
+  "title": "Compute Guard Project",
+  "metadata": {
+    "description": "Sample project for enforcing security policies on compute order service"
+  },
+  "org_id": "organization-id"
```
Comment (kushsharma) [inline cmd/seed/project.json:7]: We can leave `organization-id` empty `""`

File: cmd/seed/user.json [hunk 13]
```diff
@@ -0,0 +1,8 @@
+{
+    "name":"",
```
Comment (kushsharma) [inline cmd/seed/user.json:2]: name could be `johndoe`?

File: cmd/seed/organization.json [hunk 14]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (kushsharma) [inline cmd/seed/organization.json:1]: Let's have at least 2 organizations and 2 projects? Useful for seeing how a list looks in our UI. Also I don't see any policies being created to assign role to users.

File: internal/api/v1beta1/role.go (hunk 1 above)
Comment (Chief-Rishab) [reply internal/api/v1beta1/role.go:84]: it was already there in interceptor before this

File: internal/api/v1beta1/role.go (hunk 3 above)
Comment (Chief-Rishab) [reply internal/api/v1beta1/role.go:255]: same as above, was already there in the interceptor

File: cmd/seed/organization.json (hunk 14 above)
Comment (Chief-Rishab) [reply cmd/seed/organization.json]: done

**Review by kushsharma — APPROVED** [review]
LGTM

**Pull request merged**